components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -19,6 +19,27 @@ import org.mozilla.appservices.logins.GleanMetrics.LoginsStore as LoginsStoreMet
 import mozilla.telemetry.glean.private.CounterMetricType
 import mozilla.telemetry.glean.private.LabeledMetricType
 
+/**
+ * Typealias for backwards compatibility
+ */
+typealias LoginsStorageException = LoginsException
```

Comment:
You know I would have assumed line 25 implies all of the type aliases in lines 26 - 42 but then looking at [all of the logins type alias declarations in a-c](https://github.com/mozilla-mobile/android-components/blob/b827bd3c9c90a315001be415380e8d748894017d/components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/SyncableLoginsStorage.kt#L58) I'm guessing I'd be wrong. If the goal is to reduce breaking changes, then I suppose this is warranted. But since this PR already includes breaking changes (at least based on the change log), I'm on the fence. If we keep these nested type aliases we should at least use the naming in a-c's `SyncableLoginsStorage.kt` and remove those lines so we're only converting errors once.

Reply:
I tried some more to get this to work, but couldn't.  I think the issue is related the Kotlin -> Java translation.

I think the solution @lougeniaC64 proposed would work, but would also be a breaking change, so I we should just accept the rename.

Reply:
Also, @mhammond on another reading of your comment I'm wondering if you were asking if we need to expose all of these error types. Was that the case? If so I'm wondering what we'd expect consumers to do when errors like  `SyncAdapterError` or `SqlError` are thrown. Neither error shows up searches of a-c and iOS which seems to indicate that they're not explicitly caught. So maybe consolidating some of the non-actionable and non-fatal errors into a generic error type would be more effective? Just spitballing, mainly just wanted to make sure I didn't misinterpret and potentially bury Mark's sentiment.

Reply:
I'm pretty sure that's what @mhammond meant, and I think it's an important point regardless.  I reworked this PR to take this into account more.